vassal-app/src/main/java/VASSAL/build/module/Map.java

```diff
@@ -1971,7 +1967,25 @@ public void dragEnter(DropTargetDragEvent dtde) {
    */
   @Override
   public void dragOver(DropTargetDragEvent dtde) {
-    scrollAtEdge(dtde.getLocation(), SCROLL_ZONE);
+    final List<DataFlavor> flavors = Arrays.asList(dtde.getCurrentDataFlavors());
+
+    // Don't scroll at edge if dragging a file
+    if (flavors.contains(DataFlavor.javaFileListFlavor)) return;
+
+    if (flavors.contains(DataFlavor.stringFlavor)) {
+      final String string;
+      try {
+        string = (String) dtde.getTransferable().getTransferData(DataFlavor.stringFlavor);
+      }
+      catch (UnsupportedFlavorException | IOException e) {
+        return;
+      }
```

Comment:
Since you use this hunk of code to get the string twice, maybe make it a private member function that returns a String or null?